### PR TITLE
feat(linter): add `import/first` rule

### PR DESCRIPTION
**Pull request by pumano**

### CODE REVIEW

File: crates/oxc_linter/src/rules/import/first.rs [hunk 1]
```diff
@@ -0,0 +1,194 @@
+use oxc_ast::{ast::Statement, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule};
+
+fn first_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Using expressions between imports are prohibited")
```
Comment (pumano) [inline crates/oxc_linter/src/rules/import/first.rs:9]: Maybe "expressions" should be changed to something like "statements" or something else?

Comment (Boshen) [reply crates/oxc_linter/src/rules/import/first.rs:9]: Yeah the messages in https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/first.js is also not good.

The message should reflect where the span is pointing at.

Something like "import statements must come first"?

File: crates/oxc_linter/src/rules/import/first.rs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            return;
+        };
+        let AstKind::Program(program) = root.kind() else { unreachable!() };
+
+        for statement in &program.body {
+            match statement {
+                Statement::TSImportEqualsDeclaration(decl) => match decl.module_reference {
+                    TSModuleReference::ExternalModuleReference(ref mod_ref) => {
+                        if matches!(self.absolute_first, AbsoluteFirst::AbsoluteFirst) {
+                            if mod_ref.expression.value.as_str().starts_with("./") {
```
Comment (camchenry) [inline crates/oxc_linter/src/rules/import/first.rs:120]: 1. we should probably extract this to a `is_relative_path` helper in this file at least.
2. should we support and check for `../` here as well?

Comment (pumano) [reply crates/oxc_linter/src/rules/import/first.rs:120]: 1. Ok I will try
2. No, it's just check for relative import only for case for absolute import setting

**Review by camchenry — APPROVED** [review]

File: crates/oxc_linter/src/rules/import/first.rs [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// import { y } from 'bar';
+    /// import { x } from './foo'
+    /// ```
+    ///
+    First,
+    restriction,
```
Comment (camc314) [inline crates/oxc_linter/src/rules/import/first.rs:89]: should this be style?

File: crates/oxc_linter/src/rules/import/first.rs [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+impl AbsoluteFirst {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "absolute-first" => Self::AbsoluteFirst,
+            _ => Self::DisableAbsoluteFirst,
+        }
+    }
+}
```
Comment (camc314) [inline crates/oxc_linter/src/rules/import/first.rs:42]: should this be 

```
impl From for AbsoluteFirst...
```

https://doc.rust-lang.org/rust-by-example/conversion/from_into.html

File: crates/oxc_linter/src/rules/import/first.rs (hunk 3 above)
Comment (pumano) [reply crates/oxc_linter/src/rules/import/first.rs:89]: Maybe, if you want I can change

**Comment by pumano** [comment]
@camc314 I apply your suggested changes. Could you please look at it?

**Review by camchenry — APPROVED** [review]

**Pull request merged**